### PR TITLE
Allow null textures in setEffectTexture

### PR DESCRIPTION
**Pull request by AmoebaChant**
## Summary

- allow `setEffectTexture` callers to pass `null`
- forward `null` through the existing texture-binding cache to unbind the sampler unit
- add a regression test covering bind followed by null unbind

## Validation

- focused GL setter and cache tests: 47 passed
- GL package typecheck
- GL test typecheck
- targeted ESLint
- `git diff --check`

### CODE REVIEW

**Comment by bjsplat** [comment]
### Lite Playground - Static Site

[Open deployed site](https://liteplayground.babylonjs.com/pr/697/)

<sub>Build 20260909.28 - merge @ 9b02cf2a8169a0adcc40638e9ad693b0d4a4161e</sub>

**Comment by bjsplat** [comment]
### Lite Playground - Static Site

[Open deployed site](https://liteplayground.babylonjs.com/pr/697/)

<sub>Build 20260909.29 - merge @ 9b02cf2a8169a0adcc40638e9ad693b0d4a4161e</sub>

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
### Lab - Static Site

[Open deployed site](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260909.29/lab/index.html)

<sub>Build 20260909.29 - merge @ 9b02cf2a8169a0adcc40638e9ad693b0d4a4161e</sub>

**Comment by bjsplat** [comment]
### Lab - Static Site

[Open deployed site](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260909.31/lab/index.html)

<sub>Build 20260909.31 - merge @ 541c564995633154c32c7f3b620b21a8071ea60f</sub>

File: packages/babylon-lite-gl/src/effect.ts [hunk 1]
```diff
@@ -543,7 +543,7 @@ export function setEffectIntArray(engine: GLEngineContext, effect: GLEffect, nam
  *  is issued — that was done exactly once per program lifetime during
  *  finalization. This is the key win over Babylon's `Effect.setTexture` which
  *  re-issues the sampler binding on every call. */
-export function setEffectTexture(engine: GLEngineContext, effect: GLEffect, samplerName: string, tex: GLTexture): void {
+export function setEffectTexture(engine: GLEngineContext, effect: GLEffect, samplerName: string, tex: GLTexture | null): void {
```
[the remark this answers is not in the document]
Comment (AmoebaChant) [reply packages/babylon-lite-gl/src/effect.ts]: Updated the JSDoc to document that passing null unbinds the assigned sampler unit. Targeted ESLint, GL typecheck, and the 10 focused effect-setter tests pass.

**Comment by bjsplat** [comment]
### Lite Playground - Static Site

[Open deployed site](https://liteplayground.babylonjs.com/pr/697/)

<sub>Build 20260909.31 - merge @ 541c564995633154c32c7f3b620b21a8071ea60f</sub>
